machine_management/user_infra/adding-aws-compute-user-infra.adoc

```diff
@@ -0,0 +1,24 @@
+[id="adding-aws-compute-user-infra"]
+= Adding compute machines to AWS by using CloudFormation templates
+include::modules/common-attributes.adoc[]
+:context: adding-aws-compute-user-infra
+
+toc::[]
+
+You can add more compute machines to your {product-title} cluster on
+Amazon Web Services (AWS) that you
+created by using the sample CloudFormation templates.
+
+.Prerequisites
+
+* You installed your cluster on AWS by using the provided
+xref:../../installing/installing_aws_user_infra/installing-aws-user-infra.adoc#installing-aws-user-infra[AWS CloudFormation templates].
+* You have the JSON file and CloudFormation template that you used to create the
+worker machines during cluster installation. If you do not have these files,
+you must recreate them by following the instructions in the
+xref:../../installing/installing_aws_user_infra/installing-aws-user-infra.adoc#installing-aws-user-infra[installation procedure].
+
+
+include::modules/machine-adding-aws-compute-cloudformation.adoc[leveloffset=+1]
+
+include::modules/installation-approve-csrs.adoc[leveloffset=+1]
```

Comment:
Per my testing, after user launched a new worker machine, the machine can not listed in the ouptut of `oc get node`, unless approve the pending clieNt request CSR.
`
[root@preserve-jialiu-ansible ~]# oc get nodes
NAME                                        STATUS   ROLES    AGE    VERSION
ip-10-0-48-181.us-east-2.compute.internal   Ready    master   134m   v1.14.6+463c73f1f
ip-10-0-54-126.us-east-2.compute.internal   Ready    worker   126m   v1.14.6+463c73f1f
ip-10-0-57-129.us-east-2.compute.internal   Ready    master   134m   v1.14.6+463c73f1f
ip-10-0-65-35.us-east-2.compute.internal    Ready    worker   126m   v1.14.6+463c73f1f
ip-10-0-79-1.us-east-2.compute.internal     Ready    master   134m   v1.14.6+463c73f1f

[root@preserve-jialiu-ansible ~]# oc get csr
NAME        AGE     REQUESTOR                                                                   CONDITION
csr-dtwf2   8m26s   system:serviceaccount:openshift-machine-config-operator:node-bootstrapper   Pending

[root@preserve-jialiu-ansible ~]# oc get csr -ojson | jq -r '.items[] | select(.status == {} ) | .metadata.name' | xargs oc adm certificate approve
certificatesigningrequest.certificates.k8s.io/csr-dtwf2 approved

[root@preserve-jialiu-ansible ~]# oc get node
NAME                                        STATUS     ROLES    AGE    VERSION
ip-10-0-48-181.us-east-2.compute.internal   Ready      master   135m   v1.14.6+463c73f1f
ip-10-0-54-126.us-east-2.compute.internal   Ready      worker   127m   v1.14.6+463c73f1f
ip-10-0-57-129.us-east-2.compute.internal   Ready      master   135m   v1.14.6+463c73f1f
ip-10-0-65-35.us-east-2.compute.internal    Ready      worker   127m   v1.14.6+463c73f1f
ip-10-0-77-219.us-east-2.compute.internal   NotReady   worker   5s     v1.14.6+463c73f1f
ip-10-0-79-1.us-east-2.compute.internal     Ready      master   135m   v1.14.6+463c73f1f

[root@preserve-jialiu-ansible ~]# oc get csr
NAME        AGE     REQUESTOR                                                                   CONDITION
csr-dtwf2   9m26s   system:serviceaccount:openshift-machine-config-operator:node-bootstrapper   Approved,Issued
csr-xbnvf   7s      system:node:ip-10-0-77-219.us-east-2.compute.internal                       Pending

[root@preserve-jialiu-ansible ~]# oc get csr -ojson | jq -r '.items[] | select(.status == {} ) | .metadata.name' | xargs oc adm certificate approve
certificatesigningrequest.certificates.k8s.io/csr-xbnvf approved

[root@preserve-jialiu-ansible ~]# oc get csr
NAME        AGE   REQUESTOR                                                                   CONDITION
csr-dtwf2   10m   system:serviceaccount:openshift-machine-config-operator:node-bootstrapper   Approved,Issued
csr-xbnvf   64s   system:node:ip-10-0-77-219.us-east-2.compute.internal                       Approved,Issued

[root@preserve-jialiu-ansible ~]# oc get node
NAME                                        STATUS   ROLES    AGE    VERSION
ip-10-0-48-181.us-east-2.compute.internal   Ready    master   136m   v1.14.6+463c73f1f
ip-10-0-54-126.us-east-2.compute.internal   Ready    worker   128m   v1.14.6+463c73f1f
ip-10-0-57-129.us-east-2.compute.internal   Ready    master   136m   v1.14.6+463c73f1f
ip-10-0-65-35.us-east-2.compute.internal    Ready    worker   128m   v1.14.6+463c73f1f
ip-10-0-77-219.us-east-2.compute.internal   Ready    worker   71s    v1.14.6+463c73f1f
ip-10-0-79-1.us-east-2.compute.internal     Ready    master   136m   v1.14.6+463c73f1f
`

So I think the description in `Approving the CSRs for your machines` section need a bit adjust.

Reply:
That is correct, the node will not join the cluster until it's `system:serviceaccount` certificate is approved. Furthermore, I don't believe it will create the `system:node` certificate until after the node has joined the cluster.

Reply:
@jstuever, 
>  However, the `system.serviceaccount` CSRs have the node name hidden in the output of oc describe csr. 
Is the method that Jianlin pointed out the best way, or is there another step that I should add to clarify which certificate you're approving?

Reply:
It gets kinda messy when you are validating `system.serviceaccount` certificates. The node name is not stored directly in the CSR; instead it is part of the certificate itself. You have to describe each of them using `oc describe csr <CSR>` and look for the `Subject.Common Name` to match `system:node:<node_name>`

```
oc get csr
NAME        AGE     REQUESTOR                                                                   CONDITION
csr-6s9jv   10m     system:node:jstuev-vj98j-m-1.c.openshift-dev-installer.internal             Approved,Issued
csr-crvjt   10m     system:node:jstuev-vj98j-m-2.c.openshift-dev-installer.internal             Approved,Issued
csr-fwlbl   10m     system:serviceaccount:openshift-machine-config-operator:node-bootstrapper   Approved,Issued
csr-hscmt   10m     system:serviceaccount:openshift-machine-config-operator:node-bootstrapper   Approved,Issued
csr-kqpvl   3m54s   system:node:jstuev-vj98j-w-a-l2dz6.c.openshift-dev-installer.internal       Approved,Issued
csr-n565v   10m     system:node:jstuev-vj98j-m-0.c.openshift-dev-installer.internal             Approved,Issued
csr-n8dzm   10m     system:serviceaccount:openshift-machine-config-operator:node-bootstrapper   Approved,Issued
csr-p8rgm   3m53s   system:node:jstuev-vj98j-w-b-v8sq9.c.openshift-dev-installer.internal       Approved,Issued
csr-pfgpg   3m46s   system:serviceaccount:openshift-machine-config-operator:node-bootstrapper   Approved,Issued
csr-sbdxs   3m32s   system:node:jstuev-vj98j-w-c-4cl5k.c.openshift-dev-installer.internal       Approved,Issued
csr-v84gt   4m7s    system:serviceaccount:openshift-machine-config-operator:node-bootstrapper   Approved,Issued
csr-z8cfs   4m8s    system:serviceaccount:openshift-machine-config-operator:node-bootstrapper   Approved,Issued

oc describe csr csr-z8cfs
Name:               csr-z8cfs
Labels:             <none>
Annotations:        <none>
CreationTimestamp:  Tue, 15 Oct 2019 12:35:56 -0700
Requesting User:    system:serviceaccount:openshift-machine-config-operator:node-bootstrapper
Status:             Approved,Issued
Subject:
         Common Name:    system:node:jstuev-vj98j-w-a-l2dz6.c.openshift-dev-installer.internal
         Serial Number:  
         Organization:   system:nodes
Events:  <none>
```

Reply:
The method Jianlin works just fine if they don't care to validate the certificates before they approve them.